lib/app/custom_widgets/event_text_with_icon.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:score_live/app/custom_widgets/events_widgets/card_event_widget.dart';
+import 'package:score_live/app/custom_widgets/events_widgets/goal_event_widget.dart';
+import 'package:score_live/app/custom_widgets/events_widgets/substitution_event_widget.dart';
+import 'package:score_live/app/custom_widgets/events_widgets/var_event_widget.dart';
+import 'package:score_live/models/match_events_model/match_events_model.dart';
+import 'package:score_live/presentation/constants/app_colors.dart';
+import 'package:score_live/presentation/constants/app_const_variables.dart';
+import 'package:score_live/presentation/constants/text_styles.dart';
+
+class EventTextWithtIcon extends StatelessWidget {
+  const EventTextWithtIcon({
+    super.key,
+    this.event,
+    this.assist,
+    required this.time,
+    required this.player,
+    required this.isHomeTeam,
+  });
+
+  final MatchEventsModel? event;
+  final String? assist;
+  final String time;
+  final String player;
+  final bool isHomeTeam;
+
+  @override
+  Widget build(BuildContext context) {
+    final eventType = event?.type ?? AppConstVariables.stringPlaceholder;
+    final eventDetail = event?.detail ?? AppConstVariables.stringPlaceholder;
+    final playerIn = assist ?? AppConstVariables.stringPlaceholder;
+    return Row(
+      mainAxisAlignment: isHomeTeam == true ? MainAxisAlignment.start : MainAxisAlignment.end,
+      children: [
+        if (isHomeTeam == true) ...[
+          Text(
+            "$time'",
+            style: const CustomTextStyle(fontSize: 12, color: AppColors.inactiveTextGrey, fontWeight: FontWeight.w700),
+          ),
+          const SizedBox(width: 6),
+          if (eventType == AppConstVariables.goal)
```

Comment:
Maybe you can convert these ifs to switch statement? Looks messy tbh

Reply:
I tried to but I think I would have to either rewrite `eventType` variable in a different way or replace const variables used here with some kind of enum, I will try to work it out now

Reply:
Oh I haven't realised that's not an enum, it should be!

Reply:
Came up with something, was quite difficult considering I had to compare string with enum, so as a result I had to convert enums to strings and wrote a bunch of extra code but it works as good as before so far